File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int min = Integer.MAX_VALUE;
        int start = 0;
        int end = nums.length - 1;
        while(start <= end){
            int mid =(start + end) / 2 ;
            if(nums[mid] < min) min = nums[mid] ;
            if(nums[start] <= nums[mid]){
                min = Math.min(min,nums[start]);
                start = mid +1;
            }
            else end = mid -1;
        }
        return min;
        
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
    let start = 0;
    let end = nums.length - 1;
    let min = Number.MAX_SAFE_INTEGER;
    while(start <= end){
        let mid = parseInt((start + end) / 2);
        min = Math.min(min,nums[mid]);
        if(nums[mid] >= nums[start] ){
            min = Math.min(min,nums[start]);
            start = mid +1;
        }
        else end = mid -1;
    }
    return min;
    
};